src/core/plugin/luapi_application.h

```diff
@@ -385,28 +379,35 @@ static int applib_layerAction(lua_State* L) {
 }
 
 static void addStrokeHelper(lua_State* L, Stroke* stroke, Layer* layer) {
+    Plugin* plugin = Plugin::getPluginFromLua(L);
+    Control* ctrl = plugin->getControl();
+
     // discard any extra arguments passed in
     lua_settop(L, 1);
     luaL_checktype(L, 1, LUA_TTABLE);
 
+    // Use current settings as default if no tool options are passed
+    ToolHandler* toolHandler = ctrl->getToolHandler();
+    ToolSize currentPenSize = toolHandler->getPenSize();
+    Color currentColor = toolHandler->getColor();
+
     // Make sure we have enough points to form a stroke
     if (stroke->getPointCount() >= 2) {
         // Get the rest of the attributes. Default to something if none is provided
-        // TODO: (willnilges) Test this. Is this robust? Should we be doing this one field at a time?
         lua_getfield(L, 1, "width");
         lua_getfield(L, 1, "color");
         lua_getfield(L, 1, "fill");
         lua_getfield(L, 1, "tool");
         lua_getfield(L, 1, "lineStyle");
-        double width = luaL_optnumber(L, -5, 1.5);
-        long long int color = luaL_optinteger(L, -4, 0);
+        double width = luaL_optnumber(L, -5, currentPenSize);
+        Color color = lua_isinteger(L, -4) ? Color(lua_tointeger(L, -4)) : currentColor;
```

Comment:
Since there is a "tool" field, the user may not only add pen strokes, but also highlighter or eraser strokes. It looks wrong to use e.g. the current pen size for a highlighter stroke. You should use the specific values for the highlighter tool when `tool=="highlighter"`. Similarly with eraser strokes (and pen strokes).

Reply:
We talked about this earlier and came up with[ these sets of defaults.](https://github.com/xournalpp/xournalpp/pull/3688#issuecomment-1003541104) I've been playing around with this, trying to figure out how to use the active tool's settings as "defaults" of sorts because it seems like that'd be way better. To do that, though, I think we'd need to require a tool parameter as input.

I'm thinking it goes like this:
If the user provides no options, then the current active settings are used.
If the user provides a tool and no options, then that tool's current settings are used.
If the user provides a tool and any number of options, then their provided options override the current tool's settings.
If the user provides no tool and any number of options, **the current active settings are used.**

Reply:
Basically, you need to provide a tool, or get the default settings

Reply:
> I'm thinking it goes like this: 
If the user provides no options, then the current active settings are used. 
If the user provides a tool and no options, then that tool's current settings are used. 
If the user provides a tool and any number of options, then their provided options override the current tool's settings. 
If the user provides no tool and any number of options, **the current active settings are used.**

Using the "current active settings" has the problem, that the current active tool can be something else than pen, highlighter or eraser (e.g. the "hand" tool). With that in mind, it may be wiser to assume tool "pen" when no tool is specified, since it will be the most used one by far.

Reply:
Works for me. Let me see what I can do. I'm looking at how `getToolInfo` works (and pretty much just stealing the logic).

Reply:
So I've just about got a solution, but for whatever reason, when I set the stroke type to Eraser, instead of erasing the stroke, this happens:
![image](https://user-images.githubusercontent.com/42927786/150063657-d13c4a8f-c761-4694-b85e-53e49ded4b19.png)

Not sure what I'm doing wrong. I set the stroke type as eraser, get the width from the current settings, but still it just draws black.

Reply:
I pushed my code if you want to take a look.

Reply:
> So I've just about got a solution, but for whatever reason, when I set the stroke type to Eraser, instead of erasing the stroke, this happens: ![image](https://user-images.githubusercontent.com/42927786/150063657-d13c4a8f-c761-4694-b85e-53e49ded4b19.png)
> 
> Not sure what I'm doing wrong. I set the stroke type as eraser, get the width from the current settings, but still it just draws black.

We can only add eraser strokes in "whiteout" mode. These strokes are like pen strokes, just in White color and thick (by default). You must specify the color.
See here for the difference in the saved file. The first two eraser strokes came from a plugin. The last one was hand-drawn (in whiteout mode).

```xml
<stroke tool="eraser" color="#000000ff" width="2.00000000">150.00000000 100.00000000 154.38421326 108.76842652</stroke>
<stroke tool="eraser" color="#000000ff" width="2.00000000">170.48390270 140.96780539 200.00000000 200.00000000 80.00000000 300.00000000 300.00000000 100.00000000</stroke>
<stroke tool="eraser" color="#ffffffff" width="12.00000000">237.76500567 88.88411427 238.25880631 88.63721395 238.75260694 88.14341332 241.96231107 86.66201141 245.17201519 85.92131046 249.12242027 84.43990856 256.77633011 81.47710475 261.46743614 79.50190221 266.15854217 
77.52669967 273.81245201 74.56389586 276.03455487 73.82319491 278.75045836 72.83559364 279.49115931 72.58869332 279.98495995 72.34179300 281.21946154 71.84799237 281.46636185 71.60109205 281.96016249 71.60109205 282.94776376 71.60109205</stroke>
```

Reply:
Why is that? I haven't actually delved too deeply into the eraser source code yet, so I don't know how the normal eraser works. I'll fix what I have in the meantime.

Reply:
- The only eraser type that adds a stroke to the layer, is the "whiteout" mode.
- In "delete stroke" mode each stroke touched by the eraser is removed from the layer.
- In "standard" mode each stroke touched by the eraser is cut into two smaller pieces, that replace the original stroke.

It would be possible to use the EraseHandler's `erase` function to erase at each point  from a list of points (like for a stroke) passed via the Lua plugin API. That has nothing to do with `addStroke` though, and should not be in this PR.

Reply:
Gotcha, thanks.

So you think that those `erase` functionalities should be in other functions (and hence PRs)? Seems a bit counterintuitive to split up eraser functionality into different API calls, since those functions are grouped the same way and used the same way. Personally, I'd either add another field to this function, or split all eraser functionality into an `eraseStroke` function, including Whiteout.

Reply:
In that case it is better to split all eraser functionality into an `erase` function (including Whiteout, which will probably not be used much anyway).

Reply:
I will have to test the new API with all options, so there could be more things that need change, but right now there is nothing else I want you to change.

Reply:
I don't know if this would be helpful, but I wrote a simple Lua plugin to test the Stroke API (requires visual inspection of course :P but maybe it could be expanded to save the file then parse the contents?)

https://github.com/WillNilges/inkpath/tree/5d55e501b455d4667ba62d3c26ccc9256ecb0006/HACKING/StrokeTest